test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -261,6 +270,8 @@ private static void validateSpec(RestSpec restSpec) {
 
     @After
     public void wipeCluster() throws Exception {
+        // Running tasks might create indices and otherwise invalidate the wipe. Lets just let the finish.
+        waitForAllRunningTasksToFinish();
```

Comment:
> It'll cause the reindex request to fail, logging a warning. That seems fine.

I'm still confused. How did this cause tests to fail then?

Also, why are we kicking off a task in a rest test but never waiting for it to complete? Seems like a reindex rest test succeeding should only happen if the task completed successfully?